File: src/main/java/entity/Departure.java
package entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Departure {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer departureId;
    
    private LocalDate departDate;
    private LocalDate endDate;
    private Integer noOfDays;
    
    @ManyToOne
    @JoinColumn(name = "package_id")
    private Package tourPackage;
    
    @OneToMany(mappedBy = "departure", cascade = CascadeType.ALL)
    private List<Booking> bookings;
}

File: DotNet_Backend/SafarnamaApplication/SafarnamaApplication/Models/Departure.cs
using System;
using System.Collections.Generic;

namespace SafarnamaApplication.Models;

public partial class Departure
{
    public int DepartureId { get; set; }

    public int? PackageId { get; set; }

    public DateOnly? DepartDate { get; set; }

    public DateOnly? EndDate { get; set; }

    public int? NoOfDays { get; set; }

    public virtual ICollection<Booking> Bookings { get; set; } = new List<Booking>();

    public virtual Package? Package { get; set; }
}
